### PR TITLE
Remove specs-storage dependency, use Lotus storiface instead

### PR DESCRIPTION
**Pull request by RobQuistNL**
It looks like this specs-storage dependency is a stripped down (possibly no longer synchronized?) version of what already is in Lotus. I'm not sure why this is the case, but to me it seems like this PR is a better solution.

Best to keep it up to date by using the dependency from the Lotus project itself, IMHO.

### CODE REVIEW

File: storage/sectorblocks/sectorblocks.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func DsKeyToDealID(key datastore.Key) (uint64, error) {
 }
 
 type SectorBuilder interface {
```
Comment (nonsense) [inline storage/sectorblocks/sectorblocks.go:50]: Actually I think we are not importing this interface, but the one from Lotus.

**Comment by nonsense** [comment]
Good catch -- we should fix this.

**Comment by RobQuistNL** [comment]
These are the only places where the specs-storage are being used

File: storage/sectorblocks/sectorblocks.go (hunk 1 above)
Comment (RobQuistNL) [reply storage/sectorblocks/sectorblocks.go]: You're right, I've removed the entire file as its not being used.

**Review by jacobheun — APPROVED** [review]
Thanks for the PR! I pushed up a commit of the merge to verify CI was passing, this is gtg.

**Pull request merged**